drivers/gnss/gnss_u_blox_m10.c

```diff
@@ -0,0 +1,888 @@
+}
+
+static int ubx_m10_configure_messages(const struct device *dev)
+{
+	int ret;
+	k_spinlock_key_t key;
+	struct ubx_m10_data *data = dev->data;
+
+	key = k_spin_lock(&data->lock);
+
+	UBX_CFG_MSG_DATA_INIT(frame_data)
+
+	struct modem_ubx_script script;
+
+	ubx_m10_modem_ubx_script_fill(dev, &script, data->ubx_frame_buf, MODEM_UBX_RETRY_DEFAULT);
+
+	/* Enabling GGA, RMC and GSV messages. */
+	frame_data.rate = 1;
+	uint8_t message_enable[] = {UBX_NMEA_GGA, UBX_NMEA_RMC, UBX_NMEA_GSV};
+
+	for (int i = 0; i < sizeof(message_enable); ++i) {
+		frame_data.message_id = message_enable[i];
+		script.ubx_frame_size = ubx_create_frame(script.ubx_frame, UBX_CFG_MSG_FRM_SZ,
+			UBX_CLASS_CFG, UBX_CFG_MSG, &frame_data, UBX_CFG_MSG_PAYLOAD_SZ);
+		if (script.ubx_frame_size < 0) {
+			ret = script.ubx_frame_size;
+			goto out;
+		}
+
+		ret = ubx_m10_modem_ubx_run_script(dev, &script);
+		if (ret < 0) {
+			goto out;
+		}
+	}
+
+	/* Disabling DTM, GBS, GLL, GNS, GRS, GSA, GST, VLW, VTG and ZDA messages. */
+	frame_data.rate = 0;
+	uint8_t message_disable[] = {UBX_NMEA_DTM, UBX_NMEA_GBS, UBX_NMEA_GLL, UBX_NMEA_GNS,
+				     UBX_NMEA_GRS, UBX_NMEA_GSA, UBX_NMEA_GST, UBX_NMEA_VLW,
+				     UBX_NMEA_VTG, UBX_NMEA_ZDA};
+
+	for (int i = 0; i < sizeof(message_disable); ++i) {
+		frame_data.message_id = message_disable[i];
+		script.ubx_frame_size = ubx_create_frame(script.ubx_frame, UBX_CFG_MSG_FRM_SZ,
+			UBX_CLASS_CFG, UBX_CFG_MSG, &frame_data, UBX_CFG_MSG_PAYLOAD_SZ);
+		if (script.ubx_frame_size < 0) {
+			ret = script.ubx_frame_size;
+			goto out;
+		}
+
+		ret = ubx_m10_modem_ubx_run_script(dev, &script);
+		if (ret < 0) {
+			goto out;
+		}
+	}
+
+out:
+	k_spin_unlock(&data->lock, key);
+
+	return ret;
+}
+
+static int ubx_m10_navigation_mode_to_ubx_dynamic_model(const struct device *dev,
+							enum gnss_navigation_mode mode)
+{
+	switch (mode) {
+	case GNSS_NAVIGATION_MODE_ZERO_DYNAMICS:
+		return UBX_DYN_MODEL_STATIONARY;
+	case GNSS_NAVIGATION_MODE_LOW_DYNAMICS:
+		return UBX_DYN_MODEL_PORTABLE;
+	case GNSS_NAVIGATION_MODE_BALANCED_DYNAMICS:
+		return UBX_DYN_MODEL_AIRBONE1G;
+	case GNSS_NAVIGATION_MODE_HIGH_DYNAMICS:
+		return UBX_DYN_MODEL_AIRBONE4G;
+	default:
+		return -EINVAL;
+	}
+}
+
+static int ubx_m10_ubx_dynamic_model_to_navigation_mode(const struct device *dev,
+							enum ubx_dynamic_model dynamic_model)
+{
+	switch (dynamic_model) {
+	case UBX_DYN_MODEL_STATIONARY:
+		return GNSS_NAVIGATION_MODE_ZERO_DYNAMICS;
+	case UBX_DYN_MODEL_PORTABLE:
+		return GNSS_NAVIGATION_MODE_LOW_DYNAMICS;
+	case UBX_DYN_MODEL_AIRBONE1G:
+		return GNSS_NAVIGATION_MODE_BALANCED_DYNAMICS;
+	case UBX_DYN_MODEL_AIRBONE4G:
+		return GNSS_NAVIGATION_MODE_HIGH_DYNAMICS;
+	default:
+		return -EINVAL;
+	}
+}
+
+static int ubx_m10_set_navigation_mode(const struct device *dev, enum gnss_navigation_mode mode)
+{
+	int ret;
+	k_spinlock_key_t key;
+	struct ubx_m10_data *data = dev->data;
+
+	key = k_spin_lock(&data->lock);
+
+	UBX_CFG_NAV5_DATA_INIT(frame_data)
```

Comment:
I don't know what benefit this macro has compared to
```
	struct ubx_cfg_nav5_data frame_data;
	ubx_cfg_nav5_data_default(&frame_data);
```
IMO writing it out is clearer than the macro :) (`struct ubx_cfg_nav5_data frame_data;` should be moved to the top of the function)